python/pyspark/mllib/linalg/distributed.py

```diff
@@ -297,6 +444,20 @@ def numCols(self):
         """
         return self._java_matrix_wrapper.call("numCols")
 
+    def computeGramianMatrix(self):
+        """
+        Computes the Gramian matrix `A^T A`. Note that this cannot be
+        computed on matrices with more than 65535 columns.
```

Comment:
Thats a good question, totally reasonable to do this in a follow up PR I think since its pretty unrelated just while we are at it good to unify things while we are working on it. Just create a follow up JIRA to do this :)